### PR TITLE
Test Reorganization

### LINKED ISSUE #1039: Split tests in grammar tests and others
The set of tests is getting quite long and so tests are quite slow but come first alphabetically. It is not so pretty to rely on the order and name them _z_ as we do now.

A better solution would be to create two folders that can be run separately but that will also require some changes to the test runner I think

### PR DESCRIPTION
**Pull request by aelhamah**
This PR reorganizes the `tests/` directory.

**Description**

- Adds subdirectories `tests/test_level/`, `tests/test_snippets/`, and `tests/test_translation_level/`
- Updates `CONTRIBUTING.md` and `build-tools/github/validate-tests` with new command to run tests
- Update imports in new subdirectories to import `HedyTester` from `Tester.py` rather than `test_level_01.py` to conform with the new file structure
- Update relative paths in `tests/test_snippets/` to conform with new file structure (added extra `../`)

**Fixes #1039**

**How to test**

The new command to test is 
```bash
$ python -m pytest
```
rather than
```bash
$ python -m pytest tests/*.py
```

### CODE REVIEW

**Comment by aelhamah** [comment]
New file tree for reference:
```bash
.
├── Tester.py
├── ab_test_setup.sh
├── test_abproxying.py
├── test_distance_algo.py
├── test_dynamo.py
├── test_level
│   ├── test_level_01.py
│   ├── test_level_02.py
│   ├── test_level_03.py
│   ├── test_level_04.py
│   ├── test_level_05.py
│   ├── test_level_06.py
│   ├── test_level_07.py
│   ├── test_level_08.py
│   ├── test_level_09.py
│   ├── test_level_10.py
│   ├── test_level_11.py
│   ├── test_level_12.py
│   ├── test_level_13.py
│   ├── test_level_14.py
│   ├── test_level_15.py
│   ├── test_level_16.py
│   ├── test_level_17.py
│   └── test_level_18.py
├── test_querylog.py
├── test_snippets
│   ├── test_adventures.py
│   └── test_level_defaults.py
├── test_translation_level
│   ├── test_translation_level_01.py
│   ├── test_translation_level_02.py
│   ├── test_translation_level_03.py
│   ├── test_translation_level_04.py
│   ├── test_translation_level_05.py
│   ├── test_translation_level_06.py
│   ├── test_translation_level_07.py
│   ├── test_translation_level_08.py
│   ├── test_translation_level_09.py
│   ├── test_translation_level_10.py
│   ├── test_translation_level_11.py
│   ├── test_translation_level_12.py
│   ├── test_translation_level_13.py
│   ├── test_translation_level_14.py
│   ├── test_translation_level_15.py
│   ├── test_translation_level_16.py
│   ├── test_translation_level_17.py
│   └── test_translation_level_18.py
├── test_utils.py
└── tests_z_yamlfile.py
```

**Review by boryanagoncharenko — APPROVED** [review]

**Comment by boryanagoncharenko** [comment]
Great rearrangement! I approved and enabled auto-merge but unfortunately there are conflicts :(

**Comment by aelhamah** [comment]
Sorry about that @boryanagoncharenko ! I just merged `main` to the branch to resolve the conflict. I think it disabled the auto-merge, but it should be good to go now!

**Comment by Felienne** [comment]
Thanks a lot for the @aelhamah! I will clear out a few more PRs today before I merge this, since all PRs will get a lot of conflicts because of this one I fear.

**Review by Felienne — APPROVED** [review]
Thanks for this @aelhamah!!

**Comment by Felienne** [comment]
Ow no, the todo fix for #2350 messes up this one, can you see if you can fix the circular dependency @TiBiBa?

Also, in this form I can't run the test locally from PyCharm because it can't find Tester.py form the subdir and I like to use the IDE to run tests (and it is more beginner-friendly too)

Can we change the import in `test_level_01.py` to something like this (conceptually, this exactly is not the thing we want but I don't know the right syntax):

```
from ../Tester import HedyTester
```

**Comment by TiBiBa** [comment]
> Ow no, the todo fix for #2350 messes up this one, can you see if you can fix the circular dependency @TiBiBa?
> 
> Also, in this form I can't run the test locally from PyCharm because it can't find Tester.py form the subdir and I like to use the IDE to run tests (and it is more beginner-friendly too)
> 
> Can we change the import in `test_level_01.py` to something like this (conceptually, this exactly is not the thing we want but I don't know the right syntax):
> 
> ```
> from ../Tester import HedyTester
> ```

Will take a look!

**Comment by TiBiBa** [comment]
Not sure how to check out a PR created from a fork?

**Comment by aelhamah** [comment]
@Felienne I apologize for not checking integration with pycharm! I was originally using VS Code; however, I just changed the imports to include a `tests.` in front of `Tester`, and that appeared to fix the problem in pycharm when running 
```bash
$ python -m pytest
```

**Comment by Felienne** [comment]
Works now in PyCharm too, thanks!

File: hedy_translation.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import yaml
 from os import path
 
-KEYWORD_LANGUAGES = list(hedy.ALL_KEYWORD_LANGUAGES.keys())
+KEYWORD_LANGUAGES = ['en', 'es', 'fr', 'nl', 'nl', 'nb_NO', 'tr', 'ar', 'hi']
```
[the remark this answers is not in the document]
Comment (Felienne) [reply hedy_translation.py:8]: Thanks, this works indeed, but I made an issue to get this back in order

**Comment by Felienne** [comment]
If you feel like more chores @aelhamah... 🙏  we also have this one open: https://github.com/Felienne/hedy/issues/1931 (that has failed once but that was a merge issue not a core problem, I think)

**Comment by aelhamah** [comment]
> If you feel like more chores @aelhamah... 🙏 we also have this one open: https://github.com/Felienne/hedy/issues/1931 (that has failed once but that was a merge issue not a core problem, I think)
 
For sure! I can take a look :)

**Pull request merged**